app/Server.hs

```diff
@@ -62,6 +62,7 @@ runServer = do
               dir "get-json-data" $ look "graphName" >>= \graphName -> liftIO $ getGraphJSON graphName,
               dir "loading" $ look "size" >>= loadingResponse,
               dir "save-json" $ look "jsonData" >>= \jsonStr -> look "nameData" >>= \nameStr -> liftIO $ saveGraphJSON jsonStr nameStr,
+              dir "export-pdf" $ exportResponse "HelloWorld",
```

Comment:
1. You should extract the necessary code to generate an image into a helper function which is used by both `image` and `export-pdf` routes separately.

2. Getting the filename into the tex document is harder. A first approach would be to hardcode this name.
A second, better approach would be to capture the randomly-generated filename, and then pass it as a commandline argument to the tex file using the technique from here: http://tex.stackexchange.com/questions/243854/possible-to-compile-a-tex-file-with-command-line-arguments-passed-to-the-tex-fil.